contracts/extensions/BatchTradeExtension.sol

```diff
@@ -0,0 +1,195 @@
+/*
+    Copyright 2022 Set Labs Inc.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+    SPDX-License-Identifier: Apache License, Version 2.0
+*/
+
+pragma solidity 0.6.10;
+pragma experimental "ABIEncoderV2";
+
+import { ISetToken } from "@setprotocol/set-protocol-v2/contracts/interfaces/ISetToken.sol";
+import { ITradeModule } from "@setprotocol/set-protocol-v2/contracts/interfaces/ITradeModule.sol";
+
+import { BaseGlobalExtension } from "../lib/BaseGlobalExtension.sol";
+import { IDelegatedManager } from "../interfaces/IDelegatedManager.sol";
+import { IManagerCore } from "../interfaces/IManagerCore.sol";
+
+/**
+ * @title BatchTradeExtension
+ * @author Set Protocol
+ *
+ * Smart contract global extension which provides DelegatedManager privileged operator(s) the ability to execute a
+ * batch of trade on a DEX and the owner the ability to restrict operator(s) permissions with an asset whitelist.
+ */
+contract BatchTradeExtension is BaseGlobalExtension {
+
+    /* ============ Structs ============ */
+
+    struct TradeInfo {
+        string exchangeName;             // Human readable name of the exchange in the integrations registry
+        address sendToken;               // Address of the token to be sent to the exchange
+        uint256 sendQuantity;            // Units of token in SetToken sent to the exchange
+        address receiveToken;            // Address of the token that will be received from the exchange
+        uint256 minReceiveQuantity;      // Min units of token in SetToken to be received from the exchange
+        bytes data;                      // Arbitrary bytes to be used to construct trade call data
+    }
+
+    /* ============ Events ============ */
+
+    event BatchTradeExtensionInitialized(
+        address indexed _setToken,
+        address indexed _delegatedManager
+    );
+
+    event TradeFailed(
```

Comment:
Does front end wants some more information here (`sendToken`, `receiveToken`)? Or does index suffice?

Reply:
I think this will be OK - we can always add info here if necessary. For the moment am assuming that the ui will have the array of trades they submitted (with all relevant info) and be able to map the index to it.

(Will check with @controtie about this though...)